src/dbnode/client/client.go

```diff
@@ -20,33 +20,36 @@
 
 package client
 
-import "sync"
+import (
+	"sync"
+)
 
 type client struct {
 	sync.Mutex
 
-	opts         Options
-	newSessionFn newSessionFn
+	opts         ReplicatedOptions
+	newSessionFn newReplicatedSessionFn
 	session      AdminSession // default cached session
 }
 
-type newSessionFn func(opts Options) (clientSession, error)
+// type newReplicatedSessionFn func(ReplicatedOptions) (replicatedSession, error)
+type newReplicatedSessionFn func(ReplicatedOptions, ...replicatedSessionOption) (clientSession, error)
 
 // NewClient creates a new client
-func NewClient(opts Options) (Client, error) {
+func NewClient(opts ReplicatedOptions) (Client, error) {
```

Comment:
This is not backward compatible. 

You can keep the existing client untouched, but stack a client that is replication aware on top. This way the change is backward compatible, existing application can choose to use low level client or replication aware client (top level)... All existing use case won't have to be changed/tested, we only need to test the new client in m3db-gateway.

Reply:
Mentioned this in an another comment:
> Just wondering what the need is for ReplicationOptions, etc. Is there a way it could just be added to the existing Options/AdminOptions? It would help avoid breaking compat on the call to `NewClient(...)`.